tsdb/wlog/watcher.go

```
@@ -291,14 +291,22 @@ func (w *Watcher) Run() error {
 		return err
 	}
 
+	tail := false
+
 	level.Debug(w.logger).Log("msg", "Tailing WAL", "lastCheckpoint", lastCheckpoint, "checkpointIndex", checkpointIndex, "currentSegment", currentSegment, "lastSegment", lastSegment)
 	for !isClosed(w.quit) {
 		w.currentSegmentMetric.Set(float64(currentSegment))
 		level.Debug(w.logger).Log("msg", "Processing segment", "currentSegment", currentSegment)
 
+		// Reset the value of lastSegment, this is to avoid having to wait too long for between reads if we're reading a
+		// segment that is not the most recent segment after startup. Ignore error here since if there was a problem it would
```

What are the benefits of ignoring the error?